src/helpers/geoLocation.ts

```diff
@@ -0,0 +1,54 @@
+import cloneDeep from 'lodash/cloneDeep';
+import isEmpty from 'lodash/isEmpty';
+import set from 'set-value';
+import { FixMe } from '../util/types';
+import { getFirstMatchingKeyAndValue } from '../v0/util';
+import * as GenericFieldMappingJson from '../v0/util/data/GenericFieldMapping.json';
+
+export default class GeoLocationHelper {
+  public static getAddressKeyAndValue(message: Record<string, FixMe>): {
+    key: string;
+    value: Record<string, FixMe>;
+  } {
+    const { value, key: foundKey } = getFirstMatchingKeyAndValue(
+      message,
+      GenericFieldMappingJson.address,
+    );
+    const { key: traitsKey } = getFirstMatchingKeyAndValue(message, GenericFieldMappingJson.traits);
+    const addressKey =
+      foundKey || (traitsKey ? `${traitsKey}.address` : GenericFieldMappingJson.address[0]);
+    return { key: addressKey, value };
+  }
+
+  public static getMessageWithGeoLocationData(
```

Comment:
what do you think about name `enrichMessageAddressWithGeoData` ?

Reply:
Done